pontoon/sync/formats/xliff.py

```diff
@@ -14,107 +15,148 @@ class XLIFFEntity(VCSTranslation):
+                key = unit.getid()
+                context = unit.xmlelement.get("id")
+                source_string = str(unit.rich_source[0])
+                source_string_plural = ""
+
+                # Get the translated string for the unit. If there's no target string, this will be an empty dictionary
+                target_string = (
+                    str(unit.get_rich_target()[0]) if unit.get_rich_target() else None
+                )
+                strings = {None: target_string} if target_string else {}
+
+                # Get the unit's comments, split by newline characters
+                comments = unit.getnotes().split("\n") if unit.getnotes() else []
+
+                # Create a new XLIFFEntity from the unit
+                entity = XLIFFEntity(
+                    key,
+                    context,
+                    source_string,
+                    source_string_plural,
+                    strings,
+                    comments,
+                    order,
+                )
+                # Add the entity to the entities dictionary using its key as the dictionary key
+                self.entities[entity.key] = entity
 
     @property
     def translations(self):
-        return self.entities
+        return sorted(self.entities.values(), key=lambda e: e.order)
 
     def save(self, locale):
-        for entity in self.entities:
-            entity.sync_changes()
+        if not self.source_resource:
+            raise SyncError(
+                "Cannot save XLIFF resource {}: No source resource given.".format(
+                    self.path
+                )
+            )
+        # Open the file at the provided path
+        with open(self.source_resource.path) as f:
+            # Read the contents of the file and encode it
+            xml = f.read().encode("utf-8")
+
+            try:
+                # Parse the xml content of the file into an XLIFF file object
+                self.xliff_file = xliff.xlifffile(xml)
+            except etree.XMLSyntaxError as err:
+                # If there is an error parsing the file, raise a ParseError
+                raise ParseError(f"Failed to parse {self.source_resource.path}: {err}")
+
+            # Loop through each unit in the XLIFF file
+            for order, unit in enumerate(self.xliff_file.units):
+                # Apply any changes made to this object to the backing unit in the xliff file.
+                key = unit.getid()
+                entity = self.entities.get(key)
+                if None in entity.strings:
+                    # Store updated nodes
+                    xml = unit.xmlelement
+                    target = xml.find(unit.namespaced("target"))
+
+                    # If there's no existing target, create a new one
+                    if target is None:
+                        target = etree.SubElement(xml, unit.namespaced("target"))
+                    unit.settarget(entity.strings[None])
```

Comment:
We should set the target at the top of this `if`-block, same as in the old code, in order to keep the `target` value up-to-date.